src/components/ui/tenantList.tsx

```diff
@@ -79,19 +92,41 @@ const TenantListItem = ({ shareHouseId, tenant }: ITenantListItemProps) => {
             </DropdownMenuGroup>
           </DropdownMenuContent>
         </DropdownMenu>
-        <TenantInvitationDrawer
-          shareHouseId={shareHouseId}
-          tenant={tenant}
-          open={openEdit}
-          setOpen={setOpenEdit}
-        />
-        <DeleteConfirmationDrawer
-          id={tenant.id}
-          idType={'tenant'}
-          deleteItem={tenant.name}
-          open={openDelete}
-          setOpen={setOpenDelete}
-        />
+
+        {type === 'setup' ? (
+          <SetupTenantInvitationDrawer
+            tenant={tenant}
+            open={openEdit}
+            setOpen={setOpenEdit}
+            editTenant={onUpdate}
+            tenantData={tenantData}
+          />
+        ) : (
+          <TenantInvitationDrawer
+            shareHouseId={shareHouseId}
+            tenant={tenant}
+            open={openEdit}
+            setOpen={setOpenEdit}
+          />
+        )}
+
+        {type === 'setup' ? (
+          <SetupDeleteConfirmationDrawer
+            idType={'tenant'}
+            deleteItem={tenant.name}
+            open={openDelete}
+            setOpen={setOpenDelete}
+            onDelete={() => (onDelete ? onDelete(tenant.id) : {})}
+          />
+        ) : (
+          <DeleteConfirmationDrawer
+            id={tenant.id}
+            idType={'tenant'}
+            deleteItem={tenant.name}
+            open={openDelete}
+            setOpen={setOpenDelete}
+          />
+        )}
```

Comment:
I think the following code can be combined into one condition, so could you modify it?